compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -465,7 +465,7 @@ pub macro compile_time_machine(<$mir: lifetime, $tcx: lifetime>) {
 
     #[inline(always)]
     fn enforce_number_no_provenance(_ecx: &InterpCx<$mir, $tcx, Self>) -> bool {
-        true
+        false
```

Comment:
Was this only read during "CTFE validation", before? Could `self.ctfe_mode.is_some()` calls be moved into here instead?

Reply:
> Was this only read during "CTFE validation", before?

Yes. https://github.com/rust-lang/rust/pull/97684 made it also relevant for `read_immediate`.

> Could self.ctfe_mode.is_some() calls be moved into here instead?

No; the `self` in validation is the validation visitor, and that has the ctfe_mode -- but that doesn't exist in `read_immediate`.

Reply:
I guess alternatively in this PR we could make `enforce_number_no_provenance` again only relevant for validation, and not for `read_immediate`. That would break Miri but we don't care about Miri on the beta branch.